plenum/server/consensus/view_change_service.py

```diff
@@ -208,17 +208,7 @@ def process_new_view_message(self, msg: NewView, frm: str):
 
         logger.info("{} processing {} from {}".format(self, msg, frm))
 
-        if frm != self._data.primary_name:
-            logger.info(
-                "{} Received NewView {} for view {} from non-primary {}; expected primary {}".format(self._data.name,
-                                                                                                     msg,
-                                                                                                     self._data.view_no,
-                                                                                                     frm,
-                                                                                                     self._data.primary_name)
-            )
-            return DISCARD, "New View from non-Primary"
-
-        self._data.new_view = msg
+        self._data.new_view_votes.add_new_view(msg, frm)
```

Comment:
Do we want to return the logger.info message because it indicates that the node works incorrectly?

Reply:
I would say this can even be a warning